### DISCUSSION TITLE
Fixing CS7036 (generate constructor based on base class) should give an option to use Primary Constructor

### DISCUSSION THREAD

**Discussion by datvm** [dotnet/roslyn#74598] [Ideas]
I am not sure if this is a duplicate because I think this is a very popular scenario. Assume we have this code:

```cs
class BaseClass(CancellationToken ct)
{

}

class DerivedClass : BaseClass
{
}
```

VS would suggest to "Generate constructor 'DerivedClass(ct)'":

![image](https://github.com/user-attachments/assets/de18aeba-e607-424a-a099-e32845a41b63)

After that, VS continues to suggest to use Primary Constructor instead which gives this code:

```cs
class DerivedClass(CancellationToken ct) : BaseClass(ct)
{
}
```

I think the analyzer should merge the two steps into one?

**Comment by CyrusNajmabadi** [comment]
Yes.  it should.

**Comment by CyrusNajmabadi** [comment]
Tracked in https://github.com/dotnet/roslyn/issues/74602
